oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/query/FulltextIndex.java

```diff
@@ -429,7 +429,12 @@ public String next() {
                     readCount++;
                     if (readCount % TRAVERSING_WARNING == 0) {
                         Cursors.checkReadLimit(readCount, settings);
-                        log.warn("Index-Traversed {} nodes with filter {}", readCount, plan.getFilter());
+                        if (readCount == 2 * TRAVERSING_WARNING) {
```

Comment:
So this will print the stack trace the second time that the traverse warning message is displayed and then it will not print it again. If the idea is to print it only once, why not print it with the first message, so the first time the user is warned?

Reply:
Yes. There will be a warning (without stack trace) at 10'000 nodes read. And at 20'000, there will be a warning PLUS a stack trace.

> why not print it with the first message

My idea is that 10'000 is kind of OKish... so for those developers that see the message and fix the problem, the more "severe" stack trace is not listed. 

Another reason is that the distribution is right skewed (the median is low), so this will log less stack traces.